lib/screens/products_overview_screen.dart

```diff
@@ -2,12 +2,45 @@ import 'package:flutter/material.dart';
 
 import '../widgets/products_grid.dart';
 
-class ProductsOverviewScreen extends StatelessWidget {
+enum FilterOptions {
+  Favorites,
+  All,
+}
+
+class ProductsOverviewScreen extends StatefulWidget {
+  @override
+  _ProductsOverviewScreenState createState() => _ProductsOverviewScreenState();
+}
+
+class _ProductsOverviewScreenState extends State<ProductsOverviewScreen> {
+  var _showOnlyFavorites = false;
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(title: Text('MyShop')),
-      body: ProductsGrid(),
+      appBar: AppBar(
+        title: Text('MyShop'),
+        actions: [
+          PopupMenuButton(
+            onSelected: (FilterOptions selectedValue) {
+              setState(() {
+                if (selectedValue == FilterOptions.Favorites) {
+                  _showOnlyFavorites = true;
+                } else {
+                  _showOnlyFavorites = false;
+                }
+              });
```

Comment:
convert this to method

Reply:
@MapleSyyrup here's an example, I hope you put this in your mind

```
void showOnlyFavorites(FilterOptions selectedValue) {
setState(() {
_showOnlyFavorites = selectedValue == FilterOptions.Favorites;
});
}
```

Reply:
I will try to explain the method, I hope you could grasp.

if `selectedValue` **_is equal to_** `FilterOptions.Favorites`, then `_showOnlyFavorites` will be `true`. Otherwise, it will be `false`